### ISSUE TITLE
Make a list of models we would like to implement

### ISSUE THREAD

**Issue by istvanp1** [skpref/skpref#29]
Write down a list of models that we would like to see in this package, this is a living document.

**Comment by istvanp1** [comment]
* [ ] Thurstone Model (+ augmented with covariates)
* [x] Luce / Bradley Terry (+ augmented with covariates)
* [ ]  Plackett-Luce
* [ ] Nested Logit models
* [ ] FATE / FETA
* [ ] SVM
* [ ] Reduction wrappers

**Comment by fkiraly** [comment]
Suggestions:
* record the result in the wiki
* order the methods by scitype: which tasks do they solve?
* have a separate section for transformers, compositors, and reducers

**Comment by fkiraly** [comment]
Is this correct?

Most important tasks are:
pairwise comparison, choice(-of-1), subset choice, top-k, ranking, continuous ranking
(variants/properties are in the design documents)

Pairwise comparison:
BT, Thurstone, pairwise GLM

choice (one, subset, or top-k):
PL, nested logit, choice GLM, FATE/FETA

continuous ranking:
SVM-variants, PL, nested logit, ranking GLM, ranking neural nets

**Comment by istvanp1** [comment]
Yes, agreed with the most important tasks. Perhaps add to clarify that top-k is only different from subset choice if the top-k is ranked additionally to being chosen, which makes it a mix between subset choice and continuous ranking.

What is the difference between ranking and continuous ranking? Do we mean a poset vs. a fully ordered set (i.e. no ties in continuous ranking)?

Do we need to distinguish between choice of 1 (I assume from many) and subset choice? Arguments:
* The only reason I'm distinguishing Pairwise Comparison to subset choice is that the data can be very often represented very differently making it much easier to have this class explicitly. However, choice of 1 from many though, is likely to follow a very similar.
* Although all glm models (except from plackett-luce which is a full ranking model) have been developed as a choice of 1, so perhaps this deserves to be it's own task, but from a data and code perspective it might look exactly the same so from a coding perspective we might end up just creating a new task with a different name that is a copy of the ChoiceTask, which might just be redundant.

**Comment by fkiraly** [comment]
>What is the difference between ranking and continuous ranking?

In continuous ranking, the assigned ranks can be real numbers.

In the vanilla version, this is a version of supervised regression, where you predict a real y from X.
The goal is though not to predict y accurately, but to predict something where the ranking of the predictions, over the entire test set, is similar ("concordant") to the ranking of the real y.

This is slightly distinct from learning to rank or choice etc, since the rank is assigned to individual items, based on features - rather than starting with a bunch of objects, and ranking them in batches.
It is also slightly distinct as "true" continuous ranks are observed in vanilla continuous ranking.

It is, however, also very similar, if you consider the test set in the vanilla version of continuous ranking as a single batch.
It is also related to the labelling case that @kiudee likes.

**Comment by fkiraly** [comment]
>Do we need to distinguish between choice of 1 (I assume from many) and subset choice?

I think it makes sense to, if it is not too much of an overhead.

**Comment by istvanp1** [comment]
> In the vanilla version, this is a version of supervised regression, where you predict a real y from X.
> The goal is though not to predict y accurately, but to predict something where the ranking of the predictions, over the entire test set, is similar ("concordant") to the ranking of the real y.

Is that along the lines of the survival SVM?

**Comment by istvanp1** [comment]
> > Do we need to distinguish between choice of 1 (I assume from many) and subset choice?
> 
> I think it makes sense to, if it is not too much of an overhead.

So this should have it's own task. What should we call it? VoteTask?
I think I agree with this since there can be reductions between subset choice and 1 from many i.e. choosing [A, B] from [A, B, C, D] can be represented as:
* full: [A] from [A, B, C, D]; [B] from [B, C, D]; [B] from [A, B, C, D]; [A] from [A, C, D]
* random sequential:
    * [A] from [A, B, C, D]; [B] from [B, C, D]
    * OR [B] from [A, B, C, D]; [A] from [A, C, D]
* non conflicting [B] from [B, C, D]; [A] from [A, C, D]
